provider/ovh/ovh.go

```diff
@@ -137,18 +137,32 @@ func (p *OVHProvider) Records(ctx context.Context) ([]*endpoint.Endpoint, error)
 }
 
 // ApplyChanges applies a given set of changes in a given zone.
-func (p *OVHProvider) ApplyChanges(ctx context.Context, changes *plan.Changes) error {
+func (p *OVHProvider) ApplyChanges(ctx context.Context, changes *plan.Changes) (err error) {
 	zones, records, err := p.zonesRecords(ctx)
-	zonesChangeUniques := map[string]bool{}
 	if err != nil {
-		return err
+		return provider.NewSoftError(err)
 	}
 
-	allChanges := make([]ovhChange, 0, countTargets(changes.Create, changes.UpdateNew, changes.UpdateOld, changes.Delete))
+	zonesChangeUniques := map[string]bool{}
+
+	// Always refresh zones even in case of errors.
+	defer func() {
+		log.Debugf("OVH: %d zones will be refreshed", len(zonesChangeUniques))
+
+		eg, _ := errgroup.WithContext(ctx)
+		for zone := range zonesChangeUniques {
+			zone := zone
```

Comment:
Probably worth to add a comment why this `zone := zone` shadowing is required here. Something like

> This is necessary because the loop variable zone is reused in each iteration of the loop, and without this line, the goroutines launched by eg.Go would all reference the same zone variable

At first look, I thought is not needed

Reply:
Done